gm4_better_fire/data/better_fire/functions/pulse_check.mcfunction

```diff
@@ -1,2 +1,5 @@
 #compares assigned run tick to current tick and calls main
 execute if score current_tick gm4_clock_tick = better_fire gm4_clock_tick run function better_fire:main
+
+#arrow piercing enchanted hit particle
+execute at @e[type=minecraft:arrow,nbt=!{PierceLevel:0b}] run particle minecraft:enchanted_hit ~ ~ ~ 0 0 0 0 1 normal @a

```

Comment:
Is this really necessary? This is doing nbt serializations for every arrow, every tick, simply for some particles.

Reply:
How about tag an entity once and run the particle from that tag? If I put the tag first in the selector, it won't check for NBT.